### ISSUE TITLE
PR Web preview is not showing up in Amplify on PR creation

### ISSUE THREAD

**Issue by kegorgas** [aws-amplify/amplify-hosting#2297]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-console/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-console/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-console/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d32n3877ewg5z0

### Region

us-east-1

### Amplify Console feature

Previews

### Describe the bug

Github PR web previews are not generating on the console / admin ui when a PR is opened.

### Expected behavior

When a Github PR is opened, a web preview should be generated and built with the status displayed in the Github PR page.

### Reproduction steps

1. Open the PR in Github to a branch that has previews enabled in the Amplify console (ex: the main branch has previews enabled in Amplify, so open a PR to merge code into main)
2. Wait for the PR merge status to complete (there will be no indication of the Amplify build status)
3. Navigate to the web preview section of the Amplify console. There will be no indication of the Amplify build.

### Additional information

This started occurring after we transferred ownership of this repo from a user's account to an organization account. We also had an issue where the app would not reconnect however I was able to work around that by connecting the app to a different repo, then reconnecting it to the target repo. After I did this, the webhook was set up correctly and the Amplify app could find and pull the code from the repo for a build. However, the PR previews are now not being working when a PR is opened. I have attempted to reinstall the Amplify app (found by going into the Amplify console preview settings) and ensured that the webhook is still connected to the Github repo. Additionally, I did see the note about ensuring that the number of branches are not being met. I enabled the option to auto-disconnect branches and verified that the number of branches are not meeting the quota (the app is well under the quota, at least from what I am seeing in the Amplify CLI). Any help would be greatly appreciated in getting this resolved.

**Comment by hloriana** [comment]
Hi 👋🏽 , thanks for providing the repro details and additional context. We're going to try and reproduce the behavior. I'll reach back out as soon as I have more information.

**Comment by hloriana** [comment]
Related to: https://github.com/aws-amplify/amplify-console/issues/2256

**Comment by hloriana** [comment]
I was able to reproduce this behavior using the steps provided. A work-around can be done by deleting the webhook created in GitHub and then reconnecting the repo in the Amplify console. A fix is being rolled out for this. We will update when it is complete!🙂

**Comment by kegorgas** [comment]
Thanks so much @hloriana . I will try this work around again (although I believe I may have tried this on my end with no luck). I will report back with if it worked or not in a bit.

**Comment by kegorgas** [comment]
@hloriana Unfortunately, I was unable to get this to work using the workaround... what I did was go into the Github repo settings and delete the webhook, then go to the Amplify settings and do the reconnect repo setting. The webhook was able to successfully install into the repo. I then created a PR but it did not get picked up by the webhook (more specifically, it is not showing in the Preview area in Amplify nor is it showing the PR status badge in the PR page). 

FWIW, I do have continuous integration set up on some of the branches and if I push to the branch, it is causing it to rebuild, so it does appear that the webhook that is being installed is working.

**Comment by hloriana** [comment]
@kegorgas The team is still working on fixing this. I just repeated the behavior again and ran into the same problem. I was able to fix this by disabling previews on the branch, uninstalling the GithHub app, and then setting up previews again. I was able to trigger the preview builds successfully. Let me know if this works for you!

**Comment by kegorgas** [comment]
@hloriana Apologies I missed this update. I just ran through these steps and I'm still not able to get the PR preview to auto generate.

**Comment by ganipcanot** [comment]
Is this the appId of belonging to the organization/enterprise account? `d32n3877ewg5z0`?

**Comment by kegorgas** [comment]
@ganipcanot Yes that is the appid. And its not working for any branch. I think they are all disabled now.

**Comment by ganipcanot** [comment]
I'm trying to dig into the logs to see what happened in your case.  If you have a pull request ID, or if you can run the scenario again and provide me the appId, branch ARN, or pull request ID I can take a deeper look.

**Comment by kegorgas** [comment]
@ganipcanot I opened up a new pr just now. Here are the details:

- AppId: d32n3877ewg5z0

I'm not sure where to find the other details but I can give the names for them:
- Branch name: kyle_amp_test
- PR id (this is the GIthub id): 621

Note that the previews are not actually running (or at least doesn't appear to do so in the console), so if these ids are generated during / after the PR preview is generated, I won't be able to see them.

**Comment by ganipcanot** [comment]
Thanks - I'll see if I can take a look at what is going on here.

**Comment by MartinTechy** [comment]
Any update on this ? 

I've also tried to re-install the app, disconnect/reconnect everything but I'm still facing the same issue. 
The repo is also an organization repo

**Comment by kegorgas** [comment]
@MartinTechy They are still looking into it. Just curious, did you move the organization repo from an individual-owned repo? That is when we started noticing the issues. The repo was set up to an individual's account (not an org) and it worked great with Amplify. However, when we transferred ownership of the repo to an organization account, that is when we started seeing these issues. So curious if that is what caused the issues that you are seeing.

**Comment by MartinTechy** [comment]
@kegorgas I moved the repo from an individual account to a organization account but the previews were not enabled while it was on the individual account. So I can't really tell if this transfer is what caused the issue

**Comment by axelra82** [comment]
Also experiencing this same exact issue and have gone over the same steps, i.e. remove webhook and app from GitHub, disable in Amplify etc.

Same result. No PR previews show in the Amplify UI console.

Looking at the `Recent Deliveries` tab in the webhook (repo->settings->Webhooks) I can see that the there are deliveries happening and the response for all of them is `202`.

The PR `checks` tab just shows `Waiting for checks information...` on the PRs.

We are on an **organization repo** and it was (to my knowledge) never an "individual-owned" repo.

**Comment by kegorgas** [comment]
@hloriana  @ganipcanot Any updates on this issue? Thanks!

**Comment by hloriana** [comment]
Hi @kegorgas, apologies for the delay but we are still working on fixing this bug. I'll make sure to keep you updated on the status of this issue. We appreciate your patience! 🙂

**Comment by Narrator** [comment]
@kegorgas While we work on a permanent solution to this bug, we have gone ahead and fixed the issue for your account/app in order to unblock you. Please try testing PR previews and let us know if you're still facing issues.

Please note that one of the precursors to the issue is when you reconnect or switch repositories on your app and if you try to do that, it can cause previews to not work until we push our permanent fix.

**Comment by kegorgas** [comment]
@Narrator This seems to have resolved this specific issue for us. Thanks so much! We don't need to migrate any repos for now, so we shouldn't run into this again in the near future. 

Also, @hloriana thanks for all of your support in this.

**Comment by ivadenis** [comment]
same issue

**Comment by aherschel** [comment]
Hi, I created a new app today, and running into the same issue. Is there a mitigation to get around this issue until a fix is released?

**Comment by ivadenis** [comment]
> @kegorgas While we work on a permanent solution to this bug, we have gone ahead and fixed the issue for your account/app in order to unblock you. Please try testing PR previews and let us know if you're still facing issues.
> 
> Please note that one of the precursors to the issue is when you reconnect or switch repositories on your app and if you try to do that, it can cause previews to not work until we push our permanent fix.

@Narrator is there any way you can help me out to fix it as well? Thank you!

**Comment by Narrator** [comment]
We have rolled out a fix for this and should now be available in all regions. Please let us know if you are still seeing the issue!

**Comment by Jangholm** [comment]
Im still experiencing this issue, using a monorepo under an organization account.

**Comment by ivantichy** [comment]
The same problem here:
App Id
d2f5kmurvds6th

Region
eu-west-2

Amplify Console feature
Previews

Describe the bug
Github PR web previews are not generating on the console / admin ui when a PR is opened. Using Github private repo.

**Comment by hloriana** [comment]
@ivantichy can you try disabling previews, reconnecting your repo, and then re-enabling previews?

**Comment by jimmyn** [comment]
Hi, same problem here, have tried removing github app/webhooks/previews and re-enabling to no avail

App Id
d2q5pw1xnsetwi

Region
eu-west-1

Amplify Console feature
Previews

Describe the bug
Github PR web previews are not generating on the console / admin ui when a PR is opened. Using Github private repo in an organisation

**Comment by dadamsGS** [comment]
Same issue:
App Id
dws6rs3fo7nzq

Region
us-east-2

Amplify Console feature
Previews

Describe the bug
Github PR web previews are not generating on the console / admin ui when a PR is opened. Using Github private repo in an organization

**Comment by ciceropablo** [comment]
@hloriana, any news?

**Comment by Ad-Bean** [comment]
Same issue, please let me know if there are any updates? FYI, the pr preivews used to be okay until I deleted this app and re-created one.

App Id
d2fbku9n2b6wde

Region
us-east-1

Amplify Console feature
Previews

Describe the bug
Github PR web previews are not generating on the console / admin ui when a PR is opened. Using Github private repo in an organization

**Comment by yeezick** [comment]
Hi @hloriana, facing the same issue here. So far have tried deleting the integration, disabling previews, reconnecting the repo, re-enabling previews but no success. Please let me know how I can fix this. 
This is also on a private repo, not an organization. 

App id:
d2gsc1ze23bpdj

Region:
us-east-1

Amplify Console feature
Previews

Describe the bug
Amplify not recognizing existing PR's or new ones. Private github repo, not an organization.

**Comment by iandesj** [comment]
Also facing this issue, which is a first for the project.

App id:
d358fp3wkc3ctg

Region:
us-east-1

Amplify Console Feature:
Previews

Describe the Bug:
Amplify is not picking up my most recent pull request to deploy a preview site. This is also a private github repository in an organization.

**Comment by sanketch** [comment]
Facing this issue as well, it was working before but hasn't been now. Tried deleting the webhook, and reconnecting but still the same error. 

App id:
dfw4u0wwux0ku

Region:
us-west-1

Amplify Console Feature:
Previews

Describe the Bug:
Amplify is not picking up my most recent pull request to deploy a preview site.

**Comment by hloriana** [comment]
For folks running into this issue, we are working on root causing this behavior so we can determine a workaround. Thank you for providing your app details so we can further investigate. We will provide an update as soon as we have next steps for mitigating this behavior.

**Comment by Narrator** [comment]
@sanketch PR previews are failing for your app since you have reached the quota limit for the number of branches per app. Please increase your Amplify branch quota limit via Service Quotas: https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html

**Comment by Narrator** [comment]
@Ad-Bean Looks like your app is using a public repository. PR previews are not supported for public GitHub repositories. Please look at the following documentation: https://docs.aws.amazon.com/amplify/latest/userguide/pr-previews.html

> For security purposes, previews only work with private repositories for fullstack apps.

**Comment by Narrator** [comment]
@dadamsGS I can see that PR previews are now working for your app; Please reach out if you continue to face issues

**Comment by Narrator** [comment]
@jimmyn I can see that PR previews are now working for your app; Please reach out if you continue to face issues

**Comment by Narrator** [comment]
@rayh I can see that PR previews are now working for your app; Please reach out if you continue to face issues

**Comment by hloriana** [comment]
We are closing this issue since this bug was fixed and documentation was provided to those utilizing public repositories. If you are using a private repository and are running into this issue, please create a new GH issue so we can properly track this behavior. 

@iandesj please create a new GH issue with the bug report template if you are continuing to experience this behavior. Thank you!
